registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/cmdvalidator/DeviceValidator.java

```diff
@@ -62,81 +87,148 @@
+					responseWrapper.getErrors().get(0).getMessage());
+		}
+	}
+
+	private boolean validateTimeStamp(JSONObject payload, String packetCreationDate) throws JSONException, BaseCheckedException {
+		DateTimeFormatter format = DateTimeFormatter.ofPattern(env.getProperty(DATETIME_PATTERN));
+		LocalDateTime packetCreationDateTime = LocalDateTime
+				.parse(packetCreationDate, format);
+		LocalDateTime payloadTimestamp = LocalDateTime
+				.parse(payload.getString("timestamp"), format);
+		
+		String prefix = deviceTimeStamp.substring(0, 1);
+		String timeString = deviceTimeStamp.replaceAll("\\" + prefix, "");
+		boolean isBetween = payloadTimestamp
+				.isAfter(packetCreationDateTime.minus(Long.valueOf("2"), ChronoUnit.MINUTES))
+				&& payloadTimestamp.isBefore(
+						packetCreationDateTime.plus(Long.valueOf(timeString), ChronoUnit.MINUTES));
+		if (prefix.equals("+")) {
+			if (!isBetween) {
+				throw new BaseCheckedException(
+						StatusUtil.TIMESTAMP_AFTER_PACKETTIME.getCode(),
+						String.format(StatusUtil.TIMESTAMP_AFTER_PACKETTIME.getMessage(),
+								timeString));
+			}
+		} else if (prefix.equals("-")) {
+			if (packetCreationDateTime
+					.isBefore(payloadTimestamp.plus(Long.valueOf(timeString), ChronoUnit.MINUTES))) {
+				throw new BaseCheckedException(
+						StatusUtil.TIMESTAMP_BEFORE_PACKETTIME.getCode(),
+						String.format(StatusUtil.TIMESTAMP_BEFORE_PACKETTIME.getMessage(),
+								timeString));
+			}
+		}
+		return true;
+	}
+
+	private boolean validateSignature(JSONObject payload) throws JsonParseException, JsonMappingException, JsonProcessingException, IOException, JSONException, BaseCheckedException {
+		JWTSignatureVerifyRequestDto jwtSignatureVerifyRequestDto = new JWTSignatureVerifyRequestDto();
+		jwtSignatureVerifyRequestDto.setApplicationId("REGISTRATION");
+		jwtSignatureVerifyRequestDto.setReferenceId("SIGN");
+		jwtSignatureVerifyRequestDto.setJwtSignatureData(payload.getString("digitalId"));
+		jwtSignatureVerifyRequestDto.setValidateTrust(isTrustValidationRequired);
+		jwtSignatureVerifyRequestDto.setDomain("Device");
+		RequestWrapper<JWTSignatureVerifyRequestDto> request = new RequestWrapper<>();
+
+		request.setRequest(jwtSignatureVerifyRequestDto);
+		request.setVersion("1.0");
+		DateTimeFormatter format = DateTimeFormatter.ofPattern(env.getProperty(DATETIME_PATTERN));
+		LocalDateTime localdatetime = LocalDateTime
+				.parse(DateUtils.getUTCCurrentDateTimeString(env.getProperty(DATETIME_PATTERN)), format);
+		request.setRequesttime(localdatetime);
+		ResponseWrapper<?> responseWrapper = (ResponseWrapper<?>) registrationProcessorRestService
+				.postApi(ApiName.JWTVERIFY, "", "", request, ResponseWrapper.class);
+		if(responseWrapper.getResponse() !=null) {
+		JWTSignatureVerifyResponseDto jwtResponse = mapper.readValue(mapper.writeValueAsString(responseWrapper.getResponse()),
+				JWTSignatureVerifyResponseDto.class);
+				
+		return isTrustValidationRequired
```

Comment:
To be consistent with other validation, here also throw an exception if the validation is failed.

Reply:
I have added exception